### DISCUSSION TITLE
Dynamic / Temp Role Assignment

### DISCUSSION THREAD

**Discussion by truehybridx** [spatie/laravel-permission#2674] [Q&A]
We have use cases in applications where we can assign a role to a user permanently, but most of the time we need to be able to base their role off of calculations based on their data from external data. Is there a reasonable way we should handle this?

ie: if a user logs in and they are a supervisor (determined by other enterprise data sources queried in the moment), get get the Supervisor role and whatever permissions are assigned to that.

**Comment by drbyte** [comment]
What are you doing in your Laravel policies? What you describe sounds like something I'd be inclined to either simply add into some checks in model policies, or add some highly customized bespoke global policies that can be called instead of just checking for a specific role. (Such a custom "policy" could be just defining a new ability on the Gate which checks several roles and a few other flags ... or it could also receive data from external sources and make determinations from that. 
And then model policies could also check that custom ability if relevant, etc.)

**Reply by truehybridx** [reply]
Embarrassingly we haven't utilized policies all that much. Before looking into this package, most of the authZ was a custom route middleware checking for a explicitly assigned role in the particular application or calculated it on the spot. Messed with Gates a tiny bit here and there so I was leaning more toward that since we haven't gotten granular enough for model policies i don't think.

Right now i am trying to plan out options for handling the different use cases moving forward so our team can have better direction, though i might just be over thinking this.
